### PR TITLE
85 new post flow

### PR DESCRIPTION
**Pull request by LordofMankid**
# Description

updated the posting flow.

New Posting Flow
- added new screen for photo selection + carousel
- carousel allows for photo crop/change/removal and also a button to take more pictures
- auto navigates to gallery after taking picture or selection pictures from camera roll
- adjusted progress bar to accommodate for new screen

Edits:
- added info buttons back into the tag selection
- added correct titles in the headers of each screen
- made next step button stay in a fixed position 
- moved the camera's X 

Permissions:
- caught an issue where if users manually turns off permissions, then grant permission won't work when trying to make a post
- added back button (you'd get trapped in there) and also creates an alert to users to directly navigate to settings.

## Issues

things i didn't tackle (yet)

- didn't touch add location functionality. will be handle in issue #104 
- need to restyle search bar + spacing issues and bad/invalid location case 

- navigation/routing might be a little bit weird; if you navigate back and forth between the camera and gallery, the gesture swiping to go back might be a little off.

## Test

uhh just use it sorry can't think right now
test functionality of described features

### CODE REVIEW

**Review by johnny-t06 — APPROVED** [review]
LGTM, new posting flow is great. Small nit for animation bar but lets worry about that later. I refactored the Header component for more generality.

File: src/app/(tabs)/post/gallery.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      quality: 0.2,
+      selectionLimit: 8,
+      allowsEditing: true,
+    });
+
+    if (!result.canceled) {
+      // Loops through all of the images and sets strings of the uri
+      // TODO: make this immutable
+      handleUpdateImages(index, result.assets[0].uri);
+    }
```
[the remark this answers is not in the document]
Comment (LordofMankid) [reply src/app/(tabs)/post/gallery.tsx:47]: doesn't hurt - though i think result.cancled is when they upload 0 pics

File: src/app/(tabs)/post/gallery.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  useFocusEffect(() => {
+    updateProgress(0);
+  });
+  const handleUpdateImages = (index: number, newURI: string) => {
+    // append image links to old array
+    let updatedImageURIs = postData.imageURIs;
+
+    if (index < postData.imageURIs.length) {
+      updatedImageURIs[index] = newURI;
```
[the remark this answers is not in the document]
Comment (LordofMankid) [reply src/app/(tabs)/post/gallery.tsx:22]: isn't that covered by the else unless you mean safeguarding

**Comment by LordofMankid** [comment]
looks like the new header being position "absolute" makes styling it kind of finicky since things can go behind it, though position: "relative" introduces its own problems and i think things were going under the header anyway. lgtm but will merge tmrw

File: src/app/(tabs)/post/gallery.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (johnny-t06) [reply src/app/(tabs)/post/gallery.tsx:22]: woops yeah the else covers it

**Pull request merged**